internal/storage/metricstore/elasticsearch/query_builder.go

```diff
@@ -114,6 +115,40 @@ func (*QueryBuilder) buildTimeSeriesAggQuery(params metricstore.BaseQueryParamet
 	return dateHistAgg
 }
 
+func (*QueryBuilder) BuildAttributeValuesQuery(params *metricstore.AttributeValuesQueryParameters) (elastic.Query, elastic.Aggregation) {
+	// Create a bool query to filter by service name if provided
+	boolQuery := elastic.NewBoolQuery()
+	if params.ServiceName != "" {
+		boolQuery.Filter(elastic.NewTermQuery("process.serviceName", params.ServiceName))
+	}
+
+	// Create a global aggregation that will contain all nested aggregations
+	globalAgg := elastic.NewGlobalAggregation()
```

Comment:
why is this an aggregation query? Seems like you're doing something like `GROUP BY`, but isn't there a simpler `SELECT DISTINCT` flavor?